### DISCUSSION TITLE
Inquiry about tests

### DISCUSSION THREAD

**Discussion by siyingch86** [patrykandpatrick/vico#617] [Questions]
Hi, wonder if you have written some tests(unit test?) in this libraries and how we could access them. Thanks!

**Comment by Gowsky** [comment]
Hello, yes, we have some internal tests. However there are no public APIs for testing. Libraries don't usually bundle them. What are you looking to achieve?

**Reply by siyingch86** [reply]
I want to know if we have compose UI test like here https://developer.android.com/develop/ui/compose/testing

**Reply by Gowsky** [reply]
No, we don't have such tests. We can't use the `Modifier.semantics` API, as everything is drawn on the `Canvas` manually.

**Reply by siyingch86** [reply]
I noticed you had screenshot tests using Paparazzi.

**Reply by Gowsky** [reply]
Yes, as I said initially, we have some internal tests (including Paparazzi tests) that are not bundled with the library.
